ipatests/test_webui/test_automember.py

```diff
@@ -267,6 +354,303 @@ def test_rebuild_membership_users(self):
         self.assert_record('dev2')
 
         # Delete group, users and automember rule
-        self.delete('group', [{'pkey': 'devel'}])
-        self.delete('user', [{'pkey': 'dev1'}, {'pkey': 'dev2'}])
-        self.delete('automember', [{'pkey': 'devel'}], facet='searchgroup')
+        self.delete_user_groups('devel')
+        self.delete_users('dev1', 'dev2')
+        self.delete_user_group_rules('devel')
+
+    @screenshot
+    def test_add_multiple_user_group_rules(self):
+        """
+        Test creating and deleting multiple user group rules
+        """
+
+        groups = ['group1', 'group2', 'group3']
+
+        for group in groups:
+            self.add_user_group(group)
+
+        self.add_user_group_rules(*groups)
+        self.delete_user_group_rules(*groups)
+
+    @screenshot
+    def test_add_multiple_host_group_rules(self):
+        """
+        Test creating and deleting multiple host group rules
+        """
+
+        groups = ['group1', 'group2', 'group3']
+
+        for group in groups:
+            self.add_host_group(group)
+
+        self.add_host_group_rules(*groups)
+        self.delete_host_group_rules(*groups)
+
+    @screenshot
+    def test_search_user_group_rule(self):
+        """
+        Test searching user group rules using filter
+        """
+
+        pkey = 'search123'
+        self.add_user_group(pkey, '')
+        self.add_user_group_rules(pkey)
+
+        for text in ['search123', 'search', ' search ', 'SEARCH', '123']:
```

Comment:
does it make sense to split `' search '` into `' search'` and `'search '` i.e separate check for leading and trailing space?
IMO, in case of `' search '` only one condition will be checked (i.e either leading or trailing may be).

I would like to check behaviour with special char too like `#$2` etc.

Reply:
Yes, sure. I'll add some additional cases.

Reply:
Done.